src/webgpu/shader/execution/robust_access_vertex.spec.ts

```diff
@@ -44,7 +44,430 @@ Vertex buffer contents could be randomized to prevent the case where a previous
 a similar buffer to ours and the OOB-read seems valid. This should be deterministic, which adds
 more complexity that we may not need.`;
 
+import { params, pbool, poptions } from '../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../common/framework/test_group.js';
 import { GPUTest } from '../../gpu_test.js';
 
 export const g = makeTestGroup(GPUTest);
+
+// Arguments for a regular draw call
+enum NonIndexedDrawCallParameter {
+  VertexCount = 'VertexCount',
+  FirstVertex = 'FirstVertex',
+  InstanceCount = 'InstanceCount',
+  FirstInstance = 'FirstInstance',
+}
+
+// Arguments for an indexed draw call
+enum IndexedDrawCallParameter {
+  IndexCount = 'IndexCount',
+  FirstIndex = 'FirstIndex',
+  BaseVertex = 'BaseVertex',
+  InstanceCount = 'InstanceCount',
+  FirstInstance = 'FirstInstance',
+}
+
+// Combins both sets of parameters so we can pass one parameter 'to-be-tested'
+const DrawCallParameter = { ...NonIndexedDrawCallParameter, ...IndexedDrawCallParameter };
+type DrawCallParameter = typeof DrawCallParameter;
+const kDrawCallParameters = Object.keys(DrawCallParameter);
+
+// Encapsulates a draw call (either indexed or non-indexed)
+class DrawCall {
+  private device: GPUDevice;
+  private vertexBuffers: GPUBuffer[];
+  private indexBuffer: GPUBuffer;
+  private slotsPerBuffer: number;
+
+  // Draw
+  public vertexCount: number;
+  public firstVertex: number;
+
+  // DrawIndexed
+  public indexCount: number;
+  public firstIndex: number;
+  public baseVertex: number;
+
+  // Both Draw and DrawIndexed
+  public instanceCount: number;
+  public firstInstance: number;
+
+  constructor(
+    device: GPUDevice,
+    vertexArrays: Float32Array[],
+    vertexCount: number,
+    slotsPerBuffer: number
+  ) {
+    this.device = device;
+    this.slotsPerBuffer = slotsPerBuffer;
+    this.vertexBuffers = vertexArrays.map(v => this.GenerateVertexBuffer(v));
+
+    const indexArray = new Uint16Array(vertexCount);
+    for (let i = 0; i < vertexCount; i++) {
+      indexArray[i] = i;
+    }
+    this.indexBuffer = this.GenerateIndexBuffer(indexArray);
+
+    // Default arguments (valid call)
+    this.vertexCount = vertexCount;
+    this.firstVertex = 0;
+    this.indexCount = vertexCount;
+    this.firstIndex = 0;
+    this.baseVertex = 0;
+    this.instanceCount = vertexCount;
+    this.firstInstance = 0;
+  }
+
+  // Insert a draw call into |pass|
+  public Draw(pass: GPURenderPassEncoder) {
```

Comment:
nit: I think the CTS's naming convention is to use camelCase for function names. @kainino0x to confirm.

Reply:
Yes, and I sort of thought we had a lint rule but I guess not.

Reply:
Done